lib/Sources_process.py

```diff
@@ -221,6 +221,30 @@ def process_cve_item(self, item=None):
         if "impact" in item:
             cve["access"] = {}
             cve["impact"] = {}
+            cve['impact3'] = {}
+            cve['exploitability3'] = {}
+            if 'baseMetricV3' in item['impact']:
+                cve['impact3']['availability'] = item['impact']['baseMetricV3']['cvssV3']['availabilityImpact']
+                cve['impact3']['confidentiality'] = item['impact']['baseMetricV3']['cvssV3']['confidentialityImpact']
+                cve['impact3']['integrity'] = item['impact']['baseMetricV3']['cvssV3']['integrityImpact']
+                cve['exploitability3']['attackvector'] = item['impact']['baseMetricV3']['cvssV3']['attackVector']
+                cve['exploitability3']['attackcomplexity'] = item['impact']['baseMetricV3']['cvssV3']['attackComplexity']
+                cve['exploitability3']['privilegesrequired'] = item['impact']['baseMetricV3']['cvssV3']['privilegesRequired']
+                cve['exploitability3']['userinteraction'] = item['impact']['baseMetricV3']['cvssV3']['userInteraction']
+                cve['exploitability3']['scope'] = item['impact']['baseMetricV3']['cvssV3']['scope']
+                cve['cvss3'] = float(item['impact']['baseMetricV3']['cvssV3']['baseScore'])
+                cve['cvss3-vector'] = item['impact']['baseMetricV3']['cvssV3']['vectorString']
+            else:
+                cve['cvss3'] = None
+                cve['impact3']['availability'] = "missing V3"
+                cve['impact3']['confidentiality'] = "missing V3"
+                cve['impact3']['integrity'] = "missing V3"
+                cve['exploitability3']['attackvector'] = "missing V3"
+                cve['exploitability3']['attackcomplexity'] = "missing V3"
+                cve['exploitability3']['privilegesrequired'] = "missing V3"
+                cve['exploitability3']['userinteraction'] = "missing V3"
+                cve['exploitability3']['scope'] = "missing V3"
+                cve['cvss3-vector'] = "missing V3"
```

Comment:
I would remove this part; this clogs up the database with fields that do not actually have information in them.